File: typescript/src/giftcard/application/integration/payment/adapter/paymentRequest.ts
// Anti-Corruption Layer per tradurre gli eventi di dominio GiftCard
// nei command di integrazione esposti dal BC payment.

import type { GiftCardTopUpRequested } from '@/giftcard/domain/events/giftCardTopUpRequested.js';
import { paymentRequestIntegrationCommand } from '@/payment/integration/paymentRequestIntegrationCommand.js';
import type { PaymentRequestIntegrationCommand } from '@/payment/integration/paymentRequestIntegrationCommand.js';
import { requireArgument } from '@/common/utils/requireArgument.js';

export function paymentRequestFromTopUp(
  event: GiftCardTopUpRequested
): PaymentRequestIntegrationCommand {
  requireArgument(event, 'event');
  return paymentRequestIntegrationCommand(
    event.aggregateId.value.value,
    event.requestedAmount
  );
}

File: java/src/main/java/it/giannibombelli/wsc2026/giftcard/application/integration/payment/adapter/PaymentRequest.java
package it.giannibombelli.wsc2026.giftcard.application.integration.payment.adapter;

import it.giannibombelli.wsc2026.common.utils.Require;
import it.giannibombelli.wsc2026.giftcard.domain.events.GiftCardTopUpRequested;
import it.giannibombelli.wsc2026.payment.integration.PaymentRequestIntegrationCommand;

public final class PaymentRequest {

    private PaymentRequest() {
    }

    public static PaymentRequestIntegrationCommand fromTopUp(GiftCardTopUpRequested event) {
        Require.requireArgument(event, "event");
        return new PaymentRequestIntegrationCommand(
            event.aggregateId().value().toString(),
            event.requestedAmount()
        );
    }
}
